remill/BC/DeadStoreEliminator.cpp

```diff
@@ -0,0 +1,768 @@
+        //LOG(INFO) << LLVMThingToString(inst);
+      //}
+    //}
+    store->eraseFromParent();
+  }
+  on_remove_pass = false;
+}
+
+// Generate a DOT digraph file representing the dataflow of the LSBV.
+void LiveSetBlockVisitor::CreateDOTDigraph() {
+  auto f = func.getName().str();
+  std::ostringstream fname;
+  fname << "/tmp/out_" << f << ".dot";
+  std::ofstream dot(fname.str());
+  dot << "digraph " << f << " {\n" << "node [shape=none margin=0 nojustify=false labeljust=l]\n";
+  for (auto &block_live : block_map) {
+    auto block = block_live.first;
+    auto blive = block_live.second;
+    // top row: entry liveset
+    dot << "b" << reinterpret_cast<uintptr_t>(block) << " [label=<<table cellspacing=\"0\">\n";
+    dot << "<tr><td align=\"left\" colspan=\"3\">";
+    // print out live slots
+    auto sep = "live: ";
+    for (uint64_t i = 0; i < blive.size(); i++) {
+      if (func_used.test(i) && blive.test(i)) {
+        dot << sep << i;
+        sep = ", ";
+      }
+    }
+    dot << "</td></tr>\n";
+    for (auto &inst : *block) {
+      dot << "<tr><td align=\"left\">";
+      if (auto scope = GetScopeFromInst(inst)) {
+        auto stateslot = state_slots[scope_to_offset.at(scope)];
+        auto inst_size = 0;
+        if (auto load_inst = llvm::dyn_cast<llvm::LoadInst>(&inst)) {
+          inst_size = dl->getTypeAllocSize(load_inst->getType());
+        } else if (auto store_inst = llvm::dyn_cast<llvm::StoreInst>(&inst)) {
+          inst_size = dl->getTypeAllocSize(store_inst->getOperand(0)->getType());
+        }
+        // slot #
+        dot << "slot " << stateslot.i
+          // slot size minus load/store size
+          << "</td><td align=\"left\">" << (stateslot.size - inst_size)
+          << "</td>";
+      } else {
+        dot << "</td><td></td>";
+      }
+      // instruction text
+      if (auto call_inst = llvm::dyn_cast<llvm::CallInst>(&inst)) {
+        // calls can be quite wide, so we don't present the whole instruction
+        dot << "<td align=\"left\">call " << call_inst->getCalledValue()->getName().str();
+      } else if (llvm::isa<llvm::PHINode>(&inst)) {
+        dot << "<td align=\"left\">phi";
+      } else {
+        llvm::AAMDNodes blank(nullptr, nullptr, nullptr);
+        llvm::MDNode *tbaa = inst.getMetadata(llvm::LLVMContext::MD_tbaa);
+        llvm::MDNode *scope = inst.getMetadata(llvm::LLVMContext::MD_alias_scope);
+        llvm::MDNode *noalias = inst.getMetadata(llvm::LLVMContext::MD_noalias);
+        llvm::AAMDNodes original(tbaa, scope, noalias);
+        inst.setAAMetadata(blank);
+        if (to_remove.count(&inst)) {
+          // highlight
+          dot << "<td align=\"left\" bgcolor=\"red\">" << LLVMThingToString(&inst);
+        } else {
+          dot << "<td align=\"left\">" << LLVMThingToString(&inst);
+        }
+        inst.setAAMetadata(original);
+      }
+      dot << "</td></tr>\n";
+    }
+    // last row: exit liveset
+    LiveSet exit_live = blive;
```

Comment:
No `= blive`, `blive` represents the live set of the block on entry, so we can't just assign it to the exit live set.